### PR TITLE
docs: port is optional for socket.bind()

### PR DESCRIPTION
**Pull request by dcousens**
Same as #5686,  but I lost that fork, so I'm resubmitting it.

### CODE REVIEW

**Comment by jasnell** [comment]
LGTM

**Comment by dcousens** [comment]
Any reason not to merge?

File: doc/api/dgram.markdown [hunk 1]
```diff
@@ -156,19 +156,21 @@ a packet might travel, and that generally sending a datagram greater than
...
-and the `callback`(if specified) is called. Specifying both a
-"listening" event listener and `callback` is not harmful but not very
-useful.
+`address`.
+If `port` is not specified, the OS will try to bind to a random port.
+If `address` is not specified, the OS will try to listen on all addresses.
+After binding is done, a "listening" event is emitted and the `callback`
+(if specified) is called.
+Specifying both a "listening" event listenerand `callback` is not harmful
+but not very useful.
```
Comment (thefourtheye) [inline doc/api/dgram.markdown]: Specifying both the "listening" event listener and `callback`

File: doc/api/dgram.markdown [hunk 2]
```diff
@@ -156,19 +156,21 @@ a packet might travel, and that generally sending a datagram greater than
...
-all addresses.  After binding is done, a "listening" event is emitted
-and the `callback`(if specified) is called. Specifying both a
-"listening" event listener and `callback` is not harmful but not very
-useful.
+`address`.
+If `port` is not specified, the OS will try to bind to a random port.
+If `address` is not specified, the OS will try to listen on all addresses.
+After binding is done, a "listening" event is emitted and the `callback`
+(if specified) is called.
+Specifying both a "listening" event listenerand `callback` is not harmful
```
Comment (dcousens) [inline doc/api/dgram.markdown]: `listenerand`... might have to re-open this as a new PR,  no longer have this fork.

**Comment by jasnell** [comment]
Closing this here. Have a new PR with several doc updates bundled at http://github.com/nodejs/node

**Comment by dcousens** [comment]
Thanks @jasnell

**Pull request closed without merging**